drivers/can/Kconfig

```diff
@@ -32,6 +32,22 @@ config CAN_INIT_PRIORITY
 	  Note that the priority needs to be lower than the net stack
 	  so that it can start before the networking sub-system.
 
+config CAN_FD_MODE
+	bool
+	help
+	  Enable CAN-FD compatible API
+
+config CAN_DL
+	int "CAN-Frame Data length"
+	default 8 if !CAN_FD_MODE
+	default 64 if CAN_FD_MODE
+	range 8 64
+	help
+	  Defines the data length for CAN frames.
```

Comment:
Maybe say maximum data length for CAN frames

Reply:
Should this be user-selectable for non-CAN-FD? I think it should be hardcoded to 8 for non-CAN-FD.

Reply:
It is set to eight automatically when we have classic CAN. And the range does not allow to have less than eight.

Reply:
What I meant was that we should not expose this Kconfig setting to the user if CAN-FD is not enabled.